mixer/adapter/rbac/rbacStore.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rbac
+
+import (
+	"strings"
+
+	rbacproto "istio.io/api/rbac/v1alpha1"
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/template/authorization"
+)
+
+// Rbac interface
+type Rbac interface {
+	CheckPermission(inst *authorization.Instance, env adapter.Env) (bool, error)
+}
+
+// Information about a ServiceRole and associted ServiceRoleBindings
+type roleInfo struct {
```

Comment:
up to you:  maybe all of these types should be in a single `type()` block... and/or a `types.go`.